src/rewriter/mkrewrites.py

```diff
@@ -201,10 +167,14 @@ def validate_rule(rule):
                         die(f'List variable {child.name} cannot be used in {curr.op} and {var_to_op[child]} simultaneously'
                             )
                     var_to_op[child] = curr.op
+        elif isinstance(curr, str):
+            print(f"Unparsed string detected {curr}")
         to_visit.extend(curr.children)
 
     # Perform type checking
-    type_check(rule.lhs)
+    lhsHasConst = type_check(rule.lhs)
+    if os.getenv('CVC5_RARE_CHECK_CONST', None) is not None and lhsHasConst:
```

Comment:
Shouldn't this be an option rather than setting environment variables?

Reply:
What is the plan going forward with this check? When will it be functional enough to justify adding an option for it?

Reply:
I'm not sure about that. It would require quite a bit of context analysis

Reply:
When and where is this check used, i.e., in which use case is `CVC5_RARE_CHECK_CONST` defined?

Reply:
when you export that environment variable

Reply:
Yes, obviously, but in which scenario is this used? Is it while running regression tests like CVC5_RARE_CHECK_CONST="foo" ctest ...?

Reply:
This is only used as a convenient tool for the RARE programmer to check possible errors